packages/js/product-editor/src/components/variations-table/variations-filter/variations-filter.tsx

```diff
@@ -240,7 +240,11 @@ export function VariationsFilter( {
 									type="search"
 									value={ search }
 									suffix={ <Icon icon={ searchIcon } /> }
-									onChange={ handleInputControlChange }
+									onChange={ ( value ) => {
+										if ( value ) {
```

Comment:
Could this be handled inside of `handleInputControlChange`?

Reply:
IMHO it makes sense that this logic should be managed by the consumer, not the function, as it aligns better with separation of concerns. It is not a strong preference - what do you think?

Reply:
In this case moving this logic to the `handleInputControlChange` is still the consumer right?